File: java/05-reactor/src/test/java/com/fResult/reactor/ch5_05/TakeTest.java
package com.fResult.reactor.ch5_05;

import lombok.val;
import org.junit.jupiter.api.Test;
import reactor.core.publisher.Flux;
import reactor.test.StepVerifier;

class TakeTest {
  @Test
  void take() {
    val count = 10;
    val take = range().take(count);

    StepVerifier.create(take).expectNextCount(count).verifyComplete();
  }

  @Test
  void takeUntil() {
    val count = 50;
    val take = range().takeUntil(i -> i == (count - 1));

    StepVerifier.create(take).expectNextCount(count).verifyComplete();
  }

  private Flux<Integer> range() {
    return Flux.range(0, 1000);
  }
}

File: kotlin/05-reactor/src/test/kotlin/com/fResult/reactor/ch5_06/TakeTest.kt
package com.fResult.reactor.ch5_06

import reactor.core.publisher.Flux
import reactor.test.StepVerifier
import kotlin.test.Test

class TakeTest {
  @Test
  fun take() {
    val count = 10L
    val takenNumbers = range().take(count)
    StepVerifier.create(takenNumbers).expectNextCount(count).verifyComplete()
  }

  @Test
  fun takeUntil() {
    val count = 50L
    val takenNumbers = range().takeUntil { it.toLong() == count - 1 }
    StepVerifier.create(takenNumbers)
      .expectNextCount(count)
      .verifyComplete()
  }

  private fun range() = Flux.range(0, 1000)
}
